handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -614,21 +614,33 @@ public final SSLEngineResult wrap(
                 }
             }
 
-            if (dst.remaining() < MAX_ENCRYPTED_PACKET_LENGTH) {
+            int endOffset = offset + length;
+            long maxEncryptedLen = 0;
+
+            for (int i = offset; i < endOffset; ++i) {
+                final ByteBuffer src = srcs[i];
+                if (src == null) {
+                    throw new IllegalArgumentException("srcs[" + i + "] is null");
+                }
+                maxEncryptedLen = src.remaining();
```

Comment:
@fenik17 thanks for spotting it... https://github.com/netty/netty/pull/6252/commits/e9d9e59cb10a4258017188646f278fb7d4fba590 contains a unit test for it.